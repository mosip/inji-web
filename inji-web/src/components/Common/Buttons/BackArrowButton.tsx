import React from 'react';
import BackArrowIcon from '../../../assets/BackArrowIcon.svg';

interface BackArrowButtonProps {
    onClick: () => void;
    btnClassName?: string;
    iconClassName?: string;
    btnTestId?: string;
    iconTestId?: string;
    alt?: string;
}

export const BackArrowButton: React.FC<BackArrowButtonProps> = ({
    onClick,
    btnClassName = '',
    iconClassName = '',
    btnTestId = 'btn-back-arrow-container',
    iconTestId = 'icon-back-arrow',
    alt = 'Back Arrow'
}) => {
    return (
        <button
            type="button"
            onClick={onClick}
            data-testid={btnTestId}
            className={btnClassName}
        >
            <img
                data-testid={iconTestId}
                src={BackArrowIcon}
                alt={alt}
                className={iconClassName}
            />
        </button>
    );
};
