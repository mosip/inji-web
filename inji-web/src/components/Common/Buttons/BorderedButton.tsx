import React from "react";
import {renderGradientText} from "../../../utils/builder";

export const BorderedButton:React.FC<BorderedButtonProps> = (props) => {
  return <div className={props.fullWidth ? "w-full" : ""}>
    <div className={"bg-gradient-to-r rounded-lg text-center cursor-pointer shadow hover:shadow-lg from-iw-primary to-iw-secondary p-0.5"} onClick={props.onClick}>
            <div className={`py-1 px-2 rounded-lg bg-white justify-center center ${props.className}`}>
                <button
                    data-testid={props.testId}
                    className="text-iw-tertiary font-bold rounded-lg">
                    {props.title && renderGradientText(props.title)}
                </button>
            </div>
    </div>
  </div>
}

export type BorderedButtonProps = {
    fullWidth?: boolean | false;
    testId: string;
    onClick: (event: React.MouseEvent)=> void;
    title?: string;
    className?: string;
}
