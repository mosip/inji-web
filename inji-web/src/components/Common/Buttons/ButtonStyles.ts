export const ButtonStyles = {
    responsiveButtonWithText: {
        container: "flex items-center space-x-2 w-full",
        smallDeviceView: " w-full",
        smallDeviceButton: "sm:hidden",
        largeDeviceContainer: "hidden sm:flex items-center space-x-2",
        gradientContainer: "rounded-xl bg-gradient-to-r from-iw-primary to-iw-secondary p-[2px] transition-all duration-200 ease-in-out",
        iconButton: "flex items-center justify-center w-10 h-10 rounded-xl shadow border-none bg-[#FBF2EF] transition-all duration-200 ease-in-out",
        hoveredIconButton: "bg-gradient-to-r from-iw-primary to-iw-secondary"
    }
}