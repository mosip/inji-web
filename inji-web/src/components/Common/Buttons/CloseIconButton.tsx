import React from 'react';
import BackArrowIcon from '../../../assets/CloseIcon.svg';

interface CloseIconButtonProps {
    onClick: () => void;
    btnClassName?: string;
    iconClassName?: string;
    btnTestId?: string;
    iconTestId?: string;
    alt?: string;
}

export const CloseIconButton: React.FC<CloseIconButtonProps> = ({
    onClick,
    btnClassName = '',
    iconClassName = '',
    btnTestId = 'btn-close-container',
    iconTestId = 'icon-close',
    alt = "Close"
}) => {
    return (
        <button
            type="button"
            onClick={(e)=>{
                e.stopPropagation()
                onClick()
            }}
            data-testid={btnTestId}
            className={btnClassName}
        >
            <img
                data-testid={iconTestId}
                src={BackArrowIcon}
                alt={alt}
                className={iconClassName}
            />
        </button>
    );
};
