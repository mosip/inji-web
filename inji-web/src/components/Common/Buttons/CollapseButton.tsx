import { useSelector } from "react-redux";
import { isRTL } from "../../../utils/i18n";
import { RootState } from "../../../types/redux";
import CollapseIcon from '../../../assets/CollapseIcon.svg';

type CollapseButtonProps = {
    isCollapsed: boolean;
    onClick: () => void;
    className?: string;
};

export const CollapseButton: React.FC<CollapseButtonProps> = ({isCollapsed, onClick, className}) => {
    const language = useSelector((state: RootState) => state.common.language);

    return (
        <button
            onClick={onClick}
            className={className}
        >
            <img
                src={CollapseIcon}
                alt="Collapse"
                className={`transform ${
                    isCollapsed
                        ? isRTL(language)
                            ? ''
                            : 'rotate-180'
                        : isRTL(language)
                        ? 'rotate-180'
                        : ''
                } transition-transform duration-300 min-w-[26px] min-h-[26px]`}
            />
        </button>
    );
};