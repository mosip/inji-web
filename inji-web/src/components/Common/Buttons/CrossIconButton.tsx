import React from 'react';

interface CrossIconButtonProps {
    onClick: () => void;
    btnClassName?: string;
    iconClassName?: string;
    btnTestId?: string;
    iconTestId?: string;
}

export const CrossIconButton: React.FC<CrossIconButtonProps> = ({
                                                                    onClick,
                                                                    btnClassName = '',
                                                                    iconClassName = '',
                                                                    btnTestId = 'btn-close-icon-container',
                                                                    iconTestId = 'icon-close'
                                                                }) => {
    return (
        <button
            type="button"
            onClick={onClick}
            data-testid={btnTestId}
            className={btnClassName}
        >
            <svg
                xmlns="http://www.w3.org/2000/svg"
                viewBox="0 0 24 24"
                fill="currentColor"
                aria-hidden="true"
                className={iconClassName}
                data-testid={iconTestId}
            >
                <path
                    fill-rule="evenodd"
                    d="M6.2 6.2a1 1 0 011.4 0L12 10.6l4.4-4.4a1 1 0 111.4 1.4L13.4 12l4.4 4.4a1 1 0 01-1.4 1.4L12 13.4l-4.4 4.4a1 1 0 11-1.4-1.4L10.6 12 6.2 7.6a1 1 0 010-1.4z"
                    clip-rule="evenodd"
                />
            </svg>
        </button>
    );
};
