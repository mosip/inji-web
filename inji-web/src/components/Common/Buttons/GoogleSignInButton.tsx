import { useState } from "react";
import { FcGoogle } from "react-icons/fc";

export const GoogleSignInButton:React.FC<GoogleSignInButtonProps> = (props) => {
  const [isLoading, setIsLoading] = useState(false);
  const onClickHandler = () => {
    setIsLoading(true);
    props.handleGoogleLogin();
  };

  return (
    <button
      onClick={onClickHandler}
      disabled={isLoading}
      data-testid="google-login-button"
      className="w-full bg-white flex items-center justify-center gap-2.5 break-words py-2 px-4 shadow-md border border-gray-300 rounded-xl"
    >
      <FcGoogle size={24} className="flex-shrink-0" />
      {isLoading ? props.loadingText : props.text}
    </button>
  );
};

export type GoogleSignInButtonProps = {
    handleGoogleLogin: () => void;
    loadingText: string;
    text:string
  };