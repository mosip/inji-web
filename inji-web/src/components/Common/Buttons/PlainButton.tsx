import React from "react";
import {renderGradientText} from "../../../utils/builder";

export const PlainButton:React.FC<PlainButtonProps> = (props) => {
    return <div className={props.fullWidth ? "w-full" : ""}>
        <div className={"rounded-lg text-center cursor-pointer"} onClick={props.onClick}>
            <div className="py-2 px-4 rounded-lg bg-white">
                <button
                    data-testid={props.testId}
                    className="font-bold rounded-lg">
                    {props.disableGradient ? props.title : renderGradientText(props.title)}
                </button>
            </div>
        </div>
    </div>
}

export type PlainButtonProps = {
    fullWidth?: boolean | false;
    testId: string;
    onClick: ()=> void;
    title: string;
    disableGradient?: boolean; 
}
