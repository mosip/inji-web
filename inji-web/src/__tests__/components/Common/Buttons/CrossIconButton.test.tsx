import React from 'react';
import {render, screen, fireEvent} from '@testing-library/react';
import '@testing-library/jest-dom';
import {CrossIconButton} from '../../../../components/Common/Buttons/CrossIconButton';

jest.mock('../../../../assets/CrossIcon.svg', () => 'mocked-cross-icon-path');

describe('CrossIconButton Component', () => {
    const mockOnClick = jest.fn();

    beforeEach(() => {
        mockOnClick.mockClear();
    });

    it('matches snapshot with default props', () => {
        const {asFragment} = render(
            <CrossIconButton
                onClick={mockOnClick}
                btnTestId="btn-close-icon"
                iconTestId="icon-close"
            />
        );

        expect(asFragment()).toMatchSnapshot();
    });

    it('matches snapshot with custom props', () => {
        const {asFragment} = render(
            <CrossIconButton
                onClick={mockOnClick}
                btnClassName="custom-btn-class"
                iconClassName="custom-icon-class"
                btnTestId="custom-btn-test-id"
                iconTestId="custom-icon-test-id"
            />
        );

        expect(asFragment()).toMatchSnapshot();
    });

    it('renders button with provided btnTestId', () => {
        render(
            <CrossIconButton
                onClick={mockOnClick}
                btnTestId="test-btn-id"
                iconTestId="icon-close"
            />
        );

        expect(screen.getByTestId('test-btn-id')).toBeInTheDocument();
    });

    it('renders icon with provided iconTestId', () => {
        render(
            <CrossIconButton
                onClick={mockOnClick}
                btnTestId="btn-close-icon"
                iconTestId="test-icon-id"
            />
        );

        expect(screen.getByTestId('test-icon-id')).toBeInTheDocument();
    });

    it('renders with default test IDs when not provided', () => {
        render(
            <CrossIconButton
                onClick={mockOnClick}
                btnTestId="btn-close-icon"
                iconTestId="icon-close"
            />
        );

        expect(screen.getByTestId('btn-close-icon')).toBeInTheDocument();
        expect(screen.getByTestId('icon-close')).toBeInTheDocument();
    });

    it('calls onClick handler when clicked', () => {
        render(
            <CrossIconButton
                onClick={mockOnClick}
                btnTestId="btn-close-icon"
                iconTestId="icon-close"
            />
        );

        fireEvent.click(screen.getByTestId('btn-close-icon'));

        expect(mockOnClick).toHaveBeenCalledTimes(1);
    });

    it('applies custom button classes', () => {
        render(
            <CrossIconButton
                onClick={mockOnClick}
                btnClassName="test-custom-class"
                btnTestId="btn-close-icon"
                iconTestId="icon-close"
            />
        );

        const button = screen.getByTestId('btn-close-icon');

        expect(button).toHaveClass('test-custom-class');
    });

    it('applies custom icon classes', () => {
        render(
            <CrossIconButton
                onClick={mockOnClick}
                iconClassName="test-icon-class"
                btnTestId="btn-close-icon"
                iconTestId="icon-close"
            />
        );

        const img = screen.getByTestId('icon-close');

        expect(img).toHaveClass('test-icon-class');
    });

    it('renders the cross icon image with correct src', () => {
        render(
            <CrossIconButton
                onClick={mockOnClick}
                btnTestId="btn-close-icon"
                iconTestId="icon-close"
            />
        );

        const imgElement = screen.getByAltText('Close');

        expect(imgElement).toBeInTheDocument();
        expect(imgElement).toHaveAttribute('src', 'mocked-cross-icon-path');
    });

    it('has correct button type attribute', () => {
        render(
            <CrossIconButton
                onClick={mockOnClick}
                btnTestId="btn-close-icon"
                iconTestId="icon-close"
            />
        );

        const button = screen.getByTestId('btn-close-icon');

        expect(button).toHaveAttribute('type', 'button');
    });

    it('applies empty string as default class when no className provided', () => {
        render(
            <CrossIconButton
                onClick={mockOnClick}
                btnTestId="btn-close-icon"
                iconTestId="icon-close"
            />
        );

        const button = screen.getByTestId('btn-close-icon');
        const img = screen.getByTestId('icon-close');

        expect(button.className).toBe('');
        expect(img.className).toBe('');
    });
});
