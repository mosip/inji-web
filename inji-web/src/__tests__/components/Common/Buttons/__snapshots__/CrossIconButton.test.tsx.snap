// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CrossIconButton Component matches snapshot with custom props 1`] = `
<DocumentFragment>
  <button
    class="custom-btn-class"
    data-testid="custom-btn-test-id"
    type="button"
  >
    <svg
      class="custom-icon-class"
      data-testid="custom-icon-test-id"
      fill="currentColor"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M6.2 6.2a1 1 0 011.4 0L12 10.6l4.4-4.4a1 1 0 111.4 1.4L13.4 12l4.4 4.4a1 1 0 01-1.4 1.4L12 13.4l-4.4 4.4a1 1 0 11-1.4-1.4L10.6 12 6.2 7.6a1 1 0 010-1.4z"
        fill-rule="evenodd"
      />
    </svg>
  </button>
</DocumentFragment>
`;

exports[`CrossIconButton Component matches snapshot with default props 1`] = `
<DocumentFragment>
  <button
    class=""
    data-testid="btn-close-icon-container"
    type="button"
  >
    <svg
      class=""
      data-testid="icon-close"
      fill="currentColor"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M6.2 6.2a1 1 0 011.4 0L12 10.6l4.4-4.4a1 1 0 111.4 1.4L13.4 12l4.4 4.4a1 1 0 01-1.4 1.4L12 13.4l-4.4 4.4a1 1 0 11-1.4-1.4L10.6 12 6.2 7.6a1 1 0 010-1.4z"
        fill-rule="evenodd"
      />
    </svg>
  </button>
</DocumentFragment>
`;
